edx_rbac/utils.py

```diff
@@ -99,6 +104,13 @@ def create_role_auth_claim_for_user(user):
 
         for role_string, context in role_func(user):
             if context:
-                role_string = '{}:{}'.format(role_string, context)
-            role_auth_claim.append(role_string)
+                if isinstance(context, string_types):
+                    contextual_role = '{}:{}'.format(role_string, context)
```

Comment:
nit: it would be nice not to duplicate these two lines of formatting/appending the role data.

Reply:
Created an inner function